pandas/io/parsers.py

```diff
@@ -342,14 +342,19 @@
                          _engine_doc))
 
 _fwf_widths = """\
-colspecs : list of pairs (int, int) or 'infer'. optional
+colspecs : list of pairs (int, int) or 'infer', default 'infer'
     A list of pairs (tuples) giving the extents of the fixed-width
-    fields of each line as half-open intervals (i.e.,  [from, to[ ).
+    fields of each line as half-open intervals (i.e.,  [from, to) ).
     String value 'infer' can be used to instruct the parser to try
     detecting the column specifications from the first 100 rows of
-    the data which are not being skipped via skiprows (default='infer').
-widths : list of ints. optional
-    A list of field widths which can be used instead of 'colspecs' if
+    the data which are not being skipped via skiprows (default='infer'),
+    or by using the `infer_nrows` parameter.
+infer_nrows : int, default None
```

Comment:
can you add infer_nrows after widths. pls add a versionadded tag

Reply:
default 100

Reply:
I'm switching the places and adding a versionadded tag.

Regarding the default of `None`, I mention my [reasoning in the comment for `read_fwf`](https://github.com/pandas-dev/pandas/pull/23238/files/a35abd8900e9bff3ceb2a81cdd3aecc76375dd35#r228607376). I do state here in the docs
``If not set (or set to `None`), default behavior of 100 rows is used.``

Reply:
right but better to simply say its 100